Mathlib/Computability/EpsilonNFA.lean

```diff
@@ -124,6 +150,112 @@ theorem eval_append_singleton (x : List α) (a : α) : M.eval (x ++ [a]) = M.ste
 def accepts : Language α :=
   { x | ∃ S ∈ M.accept, S ∈ M.eval x }
 
+/-- An `M.Path` represents a traversal in `M` from a start state to an end state by following a list
+  of transitions in order. -/
+inductive Path : σ → σ → List (Option α) → Prop
+  | nil : ∀ s, Path s s []
+  | cons (t s u : σ) (a : Option α) (x : List (Option α)) :
+      t ∈ M.step s a → Path t u x → Path s u (a :: x)
+
+theorem Path_nil_eq (s t : σ) : M.Path s t [] → s = t := by
+  intro h
+  cases h
+  rfl
+
+theorem Path_singleton_iff (s t : σ) (a : Option α) : M.Path s t [a] ↔ t ∈ M.step s a:= by
```

Comment:
* Lemma names are in `snake_case`
* Arguments to `Iff`s should (generally) be implicit
* The `_eq` or `_iff` suffix in lemma names is usually redundant
```
@[simp]
theorem path_singleton {s t : σ} {a : Option α} : M.Path s t [a] ↔ t ∈ M.step s a := by
```